rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -6551,9 +6551,123 @@ private void assertRefundInProcessPegInVersionLegacy(
         Assert.assertFalse(provider.getHeightIfBtcTxhashIsAlreadyProcessed(btcTx.getHash()).isPresent());
     }
 
+    private BigInteger registerFastBridgeBtcTransaction_RSKIP293(ActivationConfig.ForBlock activations, Coin valueToSend) throws IOException, BlockStoreException, BridgeIllegalArgumentException {
+
+        Context btcContext = mock(Context.class);
+        when(btcContext.getParams()).thenReturn(bridgeConstants.getBtcParams());
+
+        Repository repository = spy(createRepository());
+        BridgeStorageProvider provider = new BridgeStorageProvider(
+                repository,
+                PrecompiledContracts.BRIDGE_ADDR,
+                bridgeConstants,
+                activations
+        );
+
+        FederationSupport federationSupportMock = mock(FederationSupport.class);
+        doReturn(provider.getNewFederationBtcUTXOs()).when(federationSupportMock).getActiveFederationBtcUTXOs();
+
+        BridgeSupport bridgeSupport = spy(new BridgeSupport(
+                bridgeConstants,
+                provider,
+                mock(BridgeEventLogger.class),
+                new BtcLockSenderProvider(),
+                new PeginInstructionsProvider(),
+                repository,
+                mock(Block.class),
+                btcContext,
+                federationSupportMock,
+                mock(BtcBlockStoreWithCache.Factory.class),
+                activations
+        ));
+
+        Federation activeFederation = getFederation(bridgeConstants, "fa03", "fa04");
+        doReturn(activeFederation).when(bridgeSupport).getActiveFederation();
+
+        doReturn(bridgeConstants.getGenesisFederation()).when(bridgeSupport).getRetiringFederation();
+
+        doReturn(true).when(bridgeSupport).validationsForRegisterBtcTransaction(any(), anyInt(), any(), any());
```

Comment:
I don't think it's a good practice to mock behavior of the class you are testing.
Instead we can help you understand how to mock the dependencies of this class to respond as you want it to. In this case this requires preparing a Bitcoin blockchain to prove your Flyover transaction is a valid Bitcoin transaction.

Reply:
Tests refactored :)!